test/Benchmarks/resnet50_bottleneck.mlir

```diff
@@ -0,0 +1,241 @@
+            ^bb0(%in: f32, %out: f32):
+            linalg.yield %in : f32
+    } -> !first_conv1x1_output_tensor_t
+
+    %5 = linalg.generic {
+            indexing_maps = [#map1, #map1], 
+            iterator_types = ["parallel", "parallel", "parallel", "parallel"]
+        } 
+        ins(%2 : !first_conv1x1_output_tensor_t) 
+        outs(%4 : !first_conv1x1_output_tensor_t) {
+            ^bb0(%in: f32, %out: f32):
+            %sum = arith.addf %in, %out : f32
+            linalg.yield %sum : f32
+    } -> !first_conv1x1_output_tensor_t
+    
+    // ReLU
+    %6 = tensor.empty() : !first_conv1x1_output_tensor_t
+    %7 = linalg.generic {
+            indexing_maps = [#map1, #map1, #map1], 
+            iterator_types = ["parallel", "parallel", "parallel", "parallel"]
+        }
+        ins(%5, %cst_9 : !first_conv1x1_output_tensor_t, !first_conv1x1_output_tensor_t)
+        outs(%6 : !first_conv1x1_output_tensor_t) {
+            ^bb0(%in1: f32, %in2: f32, %out: f32):
+                %tmp = arith.maxf %in1, %in2 : f32
+                linalg.yield %tmp : f32
+    } -> !first_conv1x1_output_tensor_t
+
+    return %7 : !first_conv1x1_output_tensor_t
+}
+
+func.func @conv2d_3x3_biasadd_relu(
+        %input : !conv3x3_input_tensor_t) -> !conv3x3_output_tensor_t {
+    %cst_0 = arith.constant 0.000000e+00 : f32
+    %cst_9 = arith.constant dense<0.000000e+00> : !conv3x3_output_tensor_t
+
+    // Conv2D trained weights
+    %filter = arith.constant dense<0.0001> : !conv3x3_filter_tensor_t
+    %bias = arith.constant dense<0.32> : !conv3x3_bias_tensor_t
+
+    // 3x3 Conv2D
+    %0 = tensor.empty() : !conv3x3_output_tensor_t
+    %1 = linalg.fill ins(%cst_0 : f32) outs(%0 : !conv3x3_output_tensor_t) -> !conv3x3_output_tensor_t
+    %2 = linalg.conv_2d_nhwc_hwcf {dilations = dense<1> : tensor<2xi64>, strides = dense<1> : tensor<2xi64>} 
+                ins(%input, %filter : !conv3x3_input_tensor_t, !conv3x3_filter_tensor_t) 
+                outs(%1 : !conv3x3_output_tensor_t) -> !conv3x3_output_tensor_t
+    
+    // BiasAdd
+    %3 = tensor.empty() : !conv3x3_output_tensor_t
+    %4 = linalg.generic {
+            indexing_maps = [#map, #map1], 
+            iterator_types = ["parallel", "parallel", "parallel", "parallel"]
+        }
+        ins(%bias : !conv3x3_bias_tensor_t)
+        outs(%3 : !conv3x3_output_tensor_t) {
+            ^bb0(%in: f32, %out: f32):
+            linalg.yield %in : f32
+    } -> !conv3x3_output_tensor_t
+
+    %5 = linalg.generic {
+            indexing_maps = [#map1, #map1], 
+            iterator_types = ["parallel", "parallel", "parallel", "parallel"]
+        } 
+        ins(%2 : !conv3x3_output_tensor_t) 
+        outs(%4 : !conv3x3_output_tensor_t) {
+            ^bb0(%in: f32, %out: f32):
+            %sum = arith.addf %in, %out : f32
+            linalg.yield %sum : f32
+    } -> !conv3x3_output_tensor_t
+
+    // ReLU
+    %6 = tensor.empty() : !conv3x3_output_tensor_t
```

Comment:
Here, too. Is this how TF lowers? If so, we may have to canonicalize to avoid these extra allocations.

Reply:
Yes, the code came from IREE's TF importer.

Reply:
Ok, try to dealloc any non-returned tensor and see if the leak stops. Try building with sanitizers locally to avoid the round trip to CI.

Reply:
Yes, that's what I am doing now. :)

Reply:
Fixed the leaks. Please check now.